src/agents/common/agent.py

```diff
@@ -0,0 +1,14 @@
+from typing import Protocol
+
+
+class Agent(Protocol):
```

Comment:
Can you add `Interface` postfix to all interfaces names everywhere? like `AgentInterface` ?

Reply:
I can do that, but it is really discouraged to do so. 
Actually Agent, KymaAgent, SupervisorAgent, KubernetesAgent really make sense if you thing from domain perspective.

Reply:
https://dev.to/alvesjessica/interfaces-in-python-not-truly-an-interface-just-a-convention-43d3

Lets atleast for interface, follow the convention of adding I prefix. PEP8 does not talk about it. I see different people are using different convention over the internet.

We do not need to do any prefixes for other types like Enums etc. I would agree with the convention of Agent  -> KymaAgent | K8sAgent if we are doing inheritance. But IMO interfaces are different and they should be clearly marked as abstract.